.github/workflows/e2e-test.yaml

```diff
@@ -0,0 +1,94 @@
+name: e2e-test
+on:
+  pull_request:
+  push:
+    paths:
+      - "**.go"
+      - "tests/**"
+    branches:
+      - main
+
+jobs:
+  tests:
+    runs-on: [ self-hosted, gen3, small ]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.19"
+
+      - name: Install dependencies
+        run: |
```

Comment:
I'd love it if (more) of these were runnable outside of CI just to be able to verify changes locally before pushing to CI, or to use in development. (it'd be fine if this were linux only,)

Reply:
I've added a rough section to the README with instructions on how to run tests (and listing dependencies).

I plan to work more on Makefile (like conversing script to make targets for deduplication) and then will have a command to install/check dependencies.

Reply:
perfect!